test/ntp_client_test.rb

```diff
@@ -0,0 +1,487 @@
+
+      subject.Read
+    end
+
+    it "loads known ntp servers and known country names" do
+      expect(subject).to receive(:GetNtpServers)
+      expect(subject).to receive(:GetCountryNames)
+
+      subject.Read
+    end
+
+    it "returns false if the ntp package is not and could not be installed" do
+      expect(Yast::PackageSystem).to receive(:CheckAndInstallPackagesInteractive)
+        .with(["ntp"]).and_return(false)
+      expect(Yast::Service).not_to receive(:Enabled)
+
+      expect(subject.Read).to eql(false)
+    end
+
+    it "checks if ntpd service is enable" do
+      expect(Yast::Service).to receive(:Enabled).with("ntpd")
+
+      subject.Read
+    end
+
+    it "checks active directory's ntp dumped data file" do
+      expect(Yast::FileUtils).to receive(:Exists).with("#{Yast::Directory.vardir}/ad_ntp_data.ycp")
+
+      subject.Read
+    end
+
+    it "reads ntp config from /etc/ntp.conf" do
+      expect(subject).to receive(:ProcessNtpConf)
+
+      subject.Read
+    end
+
+    it "reads synchronization config" do
+      expect(subject).to receive(:ReadSynchronization)
+
+      subject.Read
+    end
+
+    it "reads ntpd chroot configuration" do
+      expect(Yast::SCR).to receive(:Read)
+        .with(Yast::Path.new(".sysconfig.ntp.NTPD_RUN_CHROOTED"))
+
+      subject.Read
+    end
+
+    it "returns true if all reads were performed" do
+      expect(subject.Read).to eql(true)
+    end
+  end
+
+  describe "#Write" do
+    let(:data_dir) { File.join(File.dirname(__FILE__), "data") }
+
+    around do |example|
+      ::FileUtils.cp(File.join(data_dir, "scr_root/etc/ntp.conf.original"),
+        File.join(data_dir, "scr_root/etc/ntp.conf"))
+      change_scr_root(File.join(data_dir, "scr_root"), &example)
+    end
+
+    before do
+      allow(Yast::Mode).to receive(:normal).and_return(false)
+      allow(subject).to receive(:Abort).and_return(false)
+      allow(Yast::SCR).to receive(:Read)
+        .with(Yast::Path.new(".etc.ntp_conf.all")).and_return(nil)
+      allow(Yast::FileChanges).to receive(:StoreFileCheckSum).with("/etc/ntp.conf")
+      allow(Yast::SCR).to receive(:Write)
+        .with(Yast::Path.new(".sysconfig.network.config.NETCONFIG_NTP_POLICY"), anything)
+      allow(Yast::SCR).to receive(:Write)
+        .with(Yast::Path.new(".sysconfig.network.config"), nil)
+      allow(Yast::SCR).to receive(:Execute)
+        .with(Yast::Path.new(".target.bash"), "/sbin/netconfig update -m ntp")
+        .and_return(0)
+      allow(Yast::SCR).to receive(:Write)
+        .with(Yast::Path.new(".sysconfig.ntp.NTPD_RUN_CHROOTED"), anything)
+      allow(Yast::SCR).to receive(:Write)
+        .with(Yast::Path.new(".sysconfig.ntp"), nil)
+      allow(Yast::SCR).to receive(:Execute)
+        .with(Yast::Path.new(".target.bash"),
+          "test -e #{subject.cron_file} && rm #{subject.cron_file};")
+      allow(Yast::SuSEFirewall).to receive(:Write)
+      allow(Yast::Service).to receive(:Enable)
```

Comment:
All of them are called all the time? Maybe we could think of a better organization. But with such a long method...